### ISSUE TITLE
Support for Neo4j v3.4.0

### ISSUE THREAD

**Issue by cooperka** [neo4j-graphql/neo4j-graphql-js#53]
I just upgraded to Neo4j v3.4.0 and some queries no longer work. It looks like this lib is already being tested on v3.4.0 but tests are [failing](https://circleci.com/gh/neo4j-graphql/neo4j-graphql-js/178#tests/containers/0) with the same error I'm getting. It seems related to the use of `@cypher` directives.

Example errors, taken from my own app as well as the CI tests here:

My schema:

```js
currRequests: [Request]! @cypher(statement: "MATCH (u:User {id: $id})-[:REQUESTED]->(r:Request) RETURN r")
```

Leads to:

```console
Error: GraphQL error: Type mismatch: expected List<T> but was Any (line 1, column 159 (offset: 158))
"WITH apoc.cypher.runFirstColumn("MATCH (u:User {id: $id})-[:REQUESTED]->(r:Request) RETURN r", {id:"53514f78-fa7c-4e67-810f-ed9e39d9fff3"}, True) AS x UNWIND x AS request"
                                                                                                                                                               ^
```

integration › Handle @cypher directive on QueryType:

```console
Error: GraphQL error: Type mismatch: expected List<T> but was Any (line 1, column 154 (offset: 153))
"WITH apoc.cypher.runFirstColumn("MATCH (g:Genre) WHERE toLower(g.name) CONTAINS toLower($substring) RETURN g", {substring:"Children"}, True) AS x UNWIND x AS genre"
                                                                                                                                                          ^
```

integration › GraphQL query with @cypher directive:

```console
Error: GraphQL error: Type mismatch: expected List<T> but was Any (line 1, column 185 (offset: 184))
"MATCH (movie:Movie {title:"River Runs Through It, A"}) RETURN movie { .title ,actors: [(movie)<-[:ACTED_IN]-(movie_actors:Actor) | movie_actors { .name }] ,similar: [ movie_similar IN apoc.cypher.runFirstColumn("WITH {this} AS this MATCH (this)--(:Genre)--(o:Movie) RETURN o LIMIT {limit}", {this: movie, first: 3, offset: 0, limit: 5}, true) | movie_similar { .title }][..3] } AS movie SKIP 0"
                                                                                                                                                                                         ^
```

**Comment by johnymontana** [comment]
Thanks for reporting @cooperka 

This seems to be a bug in APOC and will need to be fixed there. I've opened this issue in APOC to track: https://github.com/neo4j-contrib/neo4j-apoc-procedures/issues/819

**Comment by cooperka** [comment]
Thanks @johnymontana! Sadly those issues doesn't look very active at all 😿 Is there anywhere (e.g. README) that documents the current Neo4j version supported by this lib? Might be good to point this out so new users aren't confounded.

**Comment by Yuri2b** [comment]
I just upgraded to 3.4.1 in Neo4j Desktop and with APOC 3.4.0.1, and this type of @cypher directive is working as expected, without "Type mismatch: expected List<T>"

**Comment by cooperka** [comment]
Thanks for the update @Yuri2b! It looks like CircleCI is already using APOC v3.4.0.1 and still failing with the same error (see `APOCH_VERSION` in https://github.com/neo4j-graphql/neo4j-graphql-js/blob/master/.circleci/config.yml). I'm not sure what the disparity might be.

**Comment by cooperka** [comment]
Verified successful. I'll let that PR automatically close this issue on merge. Thanks again @Yuri2b.

**Comment by johnymontana** [comment]
I hadn't realized this was resolved in Neo4j 3.4.1 - we were planning a workaround fix in APOC, but great to see it resolved in Neo4j. It seems one unrelated 3.4 integration test is still failing (which I'll fix tomorrow), but the others look good.
